OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,293 @@
+                        logFailedResponse(response);
+                    } else {
+                        preferences = mapResponseToPreferences(response);
+                        resultType = ResultType.SUCCESS;
+                    }
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            this.fireCallback(resultType, preferences);
+        }
+
+        private void fireCallback(ResultType result, @Nullable Preferences preferences) {
+            handler.post(() -> GetPreferencesRunnable.this.callback.run(result, preferences));
+        }
+    }
+
+    static class SetPreferencesRunnable implements Runnable {
+        private final String customerId;
+        private final PreferencesSetHandler callback;
+        private final List<PreferenceUpdate> updates;
+
+        SetPreferencesRunnable(String customerId, PreferencesSetHandler callback, List<PreferenceUpdate> updates) {
+            this.customerId = customerId;
+            this.callback = callback;
+            this.updates = updates;
+        }
+
+        @Override
+        public void run() {
+            String region = config.getRegion();
+            HttpClient httpClient = HttpClient.getInstance();
+
+            ResultType result = ResultType.ERROR;
+            try {
+                String encodedCustomerId = URLEncoder.encode(this.customerId, "UTF-8");
+                String url = "https://preference-center-" + region + ".optimove.net/api/v1/preferences?customerId=" + encodedCustomerId + "&brandGroupId=" + config.getBrandGroupId();
+                JSONArray data = mapPreferenceUpdatesToArray(updates);
+
+                try (Response response = httpClient.putSync(url, data, config.getTenantId())) {
+                    if(response.isSuccessful()) {
+                        result = ResultType.SUCCESS;
+                    };
+                }
+            } catch (JSONException | IOException e) {
+                e.printStackTrace();
+            }
+
+            this.fireCallback(result);
+        }
+
+        private void fireCallback(ResultType result) {
+            handler.post(() -> SetPreferencesRunnable.this.callback.run(result));
+        }
+    }
+
+    private static JSONArray mapPreferenceUpdatesToArray(List<PreferenceUpdate> updates) throws JSONException {
+        JSONArray updatesArray = new JSONArray();
+
+        for (int i = 0; i < updates.size(); i++) {
+            String topicId = updates.get(i).getTopicId();
+            List<Channel> channels = updates.get(i).getSubscribedChannels();
+            List<Integer> subscribedChannels = new ArrayList<>();
+            for (int j = 0; j < channels.size(); j++) {
+                subscribedChannels.add(channels.get(i).getValue());
+            }
+
+            JSONObject mappedUpdate = new JSONObject();
+            mappedUpdate.put("topicId", topicId);
+            mappedUpdate.put("subscribedChannels", subscribedChannels);
```

Comment:
🐛 

it will submit smth like

{"topicId":"9374cc58-3300-4b59-91d3-7d3905674ed1","subscribedChannels":"[489, 493]"}

subscribedChannels being a string, not array ---> will unsubscribe customer from everything